init/eessi_archdetect.sh

```diff
@@ -0,0 +1,74 @@
+#!/bin/bash
+
+# current pathways implemented in EESSI
+# x86_64/generic
+# x86_64/intel/haswell
+# x86_64/intel/skylake_avx512
+# x86_64/amd/zen2
+# x86_64/amd/zen3
+# aarch64/generic
+# ppc64le/generic
+# ppc64le/power9le
+
+ARGUMENT=${1:-none}
```

Comment:
Avoid hardcoding `/proc/cpuinfo` path and `uname -m` in script, so we can unit test it easily in CI:

```
PROC_CPUINFO=${EESSI_PROC_CPUINFO:-/proc/cpuinfo}
MACHINE_TYPE=${EESSI_MACHINE_TYPE:-$(uname -m)}
```